SU2_CFD/include/variables/CAdjEulerVariable.hpp

```diff
@@ -197,6 +175,7 @@ class CAdjEulerVariable : public CVariable {
    * \brief Get the reconstruction gradient for variables at all points.
    * \return Reference to reconstruction gradient.
    */
-  inline CVectorOfMatrix& GetGradient_Reconstruction(void) final { return Gradient_Reconstruction; }
+  inline CVectorOfMatrix& GetGradient_Reconstruction() final { return Gradient_Reconstruction; }
+  inline const CVectorOfMatrix& GetGradient_Reconstruction() const final { return Gradient_Reconstruction; }
```

Comment:
The compiler tries the non-const first, always, is "nodes" is const in that context the the const version is used and auto evaluates to "const CVectorOfMatrix". auto drops references (but not pointers) and so & needs to be added explicitly, otherwise you would get a local copy.